### DISCUSSION TITLE
Undefined method GET when trying to use metadata or pagy_next_url

### DISCUSSION THREAD

**Discussion by archonic** [ddnexus/pagy#508] [Q&A]
Using version 5.10. I would look at the documentation for that version but it doesn't appear to be available anymore.

I'm implementing infinite scroll in the style of this article: https://www.stefanwienert.de/blog/2021/04/17/endless-scroll-with-turbo-streams/

Whenever I render `<%= pagy_nav(@pagy) %>`, I'm getting `NoMethodError: undefined method 'GET' for #<PurchaseRequest ..>` where PurchaseRequest is the model being paginated. I'm also getting that error when trying to use `pagy_next_url(@pagy)`. There's also no backtrace.

Do I need to require or include parts of Pagy in the view component rb to make these methods available?

**Comment by ddnexus** [comment]
From what you said, it doesn't look related with any missing Pagy method. How did you manage to suppress the backtrace?

**Reply by archonic** [reply]
It's a mystery. I'll be upgrading the app to 6.0 soon and I'll see if that's still hitting an error on `pagy_nav(@pagy)`.

**Comment by benkoshy** [comment]
https://www.stefanwienert.de/blog/2021/04/17/endless-scroll-with-turbo-streams/

There has been a PR in Turbo since which allows you to [stream on GET requests](https://turbo.hotwired.dev/handbook/streams#streaming-from-http-responses). That should simplify the stimulus controller somewhat:

**Reply by archonic** [reply]
I'm not sure what you mean. It's already performing a GET request and streaming the next page.

I modified it a bit to not have to rely on pagy nav. It just uses a scaffold URL, current page and total pages stimulus values to know if it should load the next page and load the correct URL if so.

**Reply by benkoshy** [reply]
> I'm not sure what you mean. It's already performing a GET request and streaming the next page.

your stimulus controller could be simplified.

I'd be curious to see your implementation if you are able to share it? No issues if this is not possible.
